### ISSUE TITLE
Invalid FileDescriptor causing infinite loop

### ISSUE THREAD

**Issue by blucas** [netty/netty#3443]
Netty version: 5.0.0-Alpha2-SNAPSHOT

Attention: @normanmaurer 

When my netty client attempts to connect to a server that is currently shut down, the client is correctly informed that a connection could not be established, but the following error occurs infinitely in the pipeline.  I believe there is a continuous loop occurring in ```AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady``` [here](https://github.com/netty/netty/blob/master/transport-native-epoll/src/main/java/io/netty/channel/epoll/AbstractEpollStreamChannel.java#L648). Netty is rescheduling a epollInReady() call when it should not.  The ```Throwable``` is an ```IllegalStateException``` due to an invalid file descriptor, so the next execution of ```epollInReady``` will fail, thus entering the ```catch``` clause again, and again rescheduling ```epollInReady```.  I know what the bug is, but I don't know the *correct* way to fix it.

**Edit:** What is really interesting is how was ```epollInReady``` ever actually called to begin with? The server was down, my client tried to connect and received an unsuccessful ```ChannelFuture```. So what made Netty call ```epollInReady```?  Very strange

```java
} catch (Throwable t) {
    boolean closed = handleReadException(pipeline, byteBuf, t, close);
    // closed == false
    if (!closed) {
        // trigger a read again as there may be something left to read and because of epoll ET we
        // will not get notified again until we read everything from the socket

        // Netty is erroneously rescheduling a call to epollInReady when the 
        // Throwable is due to an invalid file descriptor
        eventLoop().execute(new Runnable() {
            @Override
            public void run() {
                epollInReady();
            }
        });
    }
}
```

My client is using Epoll ET.
```
09:44:42,091 WARN  [DefaultChannelPipeline] [epollEventLoopGroup-17-0] An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
java.lang.IllegalStateException: invalid file descriptor
        at io.netty.channel.unix.FileDescriptor$1.intValue(FileDescriptor.java:41)
        at io.netty.channel.epoll.AbstractEpollChannel.doReadBytes(AbstractEpollChannel.java:232)
        at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:603)
        at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe$3.run(AbstractEpollStreamChannel.java:651)
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:352)
        at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:254)
        at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:122)
        at io.netty.util.internal.chmv8.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1412)
        at io.netty.util.internal.chmv8.ForkJoinTask.doExec(ForkJoinTask.java:280)
        at io.netty.util.internal.chmv8.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:877)
        at io.netty.util.internal.chmv8.ForkJoinPool.scan(ForkJoinPool.java:1706)
        at io.netty.util.internal.chmv8.ForkJoinPool.runWorker(ForkJoinPool.java:1661)
        at io.netty.util.internal.chmv8.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:126)
```

**Comment by normanmaurer** [comment]
will check

**Comment by normanmaurer** [comment]
Fixed in 4.0 (960a486f13c6101ed76dfb99959501995697a01e), 4.1(5bb1377072d762f446048943f201c68b64161a30) and master (c90137f1c4859ec90e073c1b62e12844d8228d42)
